### PR TITLE
Web api

### PR DESCRIPTION
**Pull request by Blawal-SE**

### CODE REVIEW

File: jquery/Script/WebApiAjaxCall.js [hunk 1]
```diff
@@ -13,104 +11,145 @@ function ajaxCall(url,type,data){
...
 
-// if(type.toLowerCase()=='post')
-// {
 //     $.ajax({
 //         type: type,
-//         async: false,  
+//        async:false, 
 //         url: url,
+//         data:data,
 //         //contentType: 'application/json; charset=UTF-8',  //send type of data to sever
```
Comment (Shanfiaz) [inline jquery/Script/WebApiAjaxCall.js:51]: Remove the commented code

File: jquery/Script/WebApiAjaxCall.js [hunk 2]
```diff
@@ -13,104 +11,145 @@ function ajaxCall(url,type,data){
...
-           
+        success:successp,
+      
+        error:errorp
+     });
+   
 
+     
+}
+function serverCallStudent(url,type,data,successp,errorp){
```
Comment (Shanfiaz) [inline jquery/Script/WebApiAjaxCall.js:22]: Why we have tow functions

File: jquery/Script/WebApiAjaxCall.js [hunk 3]
```diff
@@ -13,104 +11,145 @@ function ajaxCall(url,type,data){
...
-//             Dob: data.Dob,
-//             Password: data.Password,
-//             ConfirmPassword: data.ConfirmPassword
 
-//         },
-//         success: function (response, statusText, xhr) {
+// // if(type.toLowerCase()=='post')
+// // {
+// //     $.ajax({
+// //         type: type,
```
Comment (Shanfiaz) [inline jquery/Script/WebApiAjaxCall.js:78]: Remove the commented code

File: SchoolApplication/School.Repository/BaseRepository.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+      
+        public void Save()
+        {
+            _context.SaveChanges();
+        }
+
+        public  void Update(T obj)
+        {
+            // _context.Entry(obj).Entity = EntityState.Modified;
```
Comment (Nauman7) [inline SchoolApplication/School.Repository/BaseRepository.cs:64]: remove commented code, or add a comment to explain why we need this

File: SchoolApplication/School.Repository/BaseRepository.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        {
+            try
+            {
+                _dbset.Add(obj);
+                Save();
+                
+            }
+            catch (Exception e)
+            {
+
```
Comment (Nauman7) [inline SchoolApplication/School.Repository/BaseRepository.cs:35]: too many empty lines

File: SchoolApplication/School.Repository/StudentRepository.cs [hunk 6]
```diff
@@ -1,119 +1,106 @@
...
-                using (var context = new SchoolContext())
+                using (SchoolContext db = new SchoolContext())
                 {
-                    var students = context.Students.ToList();
-                    return students;
-                }
-            }
-            catch (Exception e)
-            {
+                    var StdList=db.Students.ToList();
```
Comment (Nauman7) [inline SchoolApplication/School.Repository/StudentRepository.cs:24]: why not using the baseRepo methods?

File: jquery/Script/StudentManagerJquery.js [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+            "ConfrimPassword": confirmPassword
+        }
+        var students = JSON.parse(localStorage.getItem("student"));
+
+        students.forEach(function(value, index) {
+            if (value.Id == student.Id) {
+               
+                students[index] = student;
+                localStorage.setItem("student", JSON.stringify(students));
+                window.location.href = 'C:\Users\Administrator\Desktop\Internship\StudentList.html';
```
Comment (Nauman7) [inline jquery/Script/StudentManagerJquery.js:370]: why giving absolute path instead of relative

File: jquery/Script/StudentManagerJquery.js [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+        $('#stdList').html('').append(appendRow);
+        $('.editNewWindowBtn').click(function () {
+         var id=$(this).attr('data-id');
+         localStorage.setItem('StudentId',id);
+         window.location.href='EditStudent.html';
+        });
+        $('.deleteBtn').click(function () {
+            debugger;
+            var id = parseInt($(this).attr('data-id'));
+            var url = baseUrl + 'Student?id=' + id;
```
Comment (Nauman7) [inline jquery/Script/StudentManagerJquery.js:33]: move baseUrl into server methods and concatenate there. No need to send from each method and its obvious

**Review by Nauman7 — CHANGES_REQUESTED** [review]

**Pull request merged**